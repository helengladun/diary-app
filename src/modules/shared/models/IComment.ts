export interface IComment {
    id: number,
    description: string,
    post_id: number
}