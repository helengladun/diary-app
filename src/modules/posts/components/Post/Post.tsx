import React, {Component} from 'react';
import {IPost} from "../../../shared/models/IPost";
import classNames from 'classnames';

interface IProps {
    data: IPost,
    chooseHandler: Function,
    removeHandler: Function,
    comments_count: number,
    isChecked: boolean
}

export class Post extends Component<IProps> {
    static defaultProps = {
        isChecked: false
    };

    constructor(props: IProps) {
        super(props);

        this.clickHandler = this.clickHandler.bind(this);
        this.removeHandler = this.removeHandler.bind(this);
    }

    clickHandler() {
        this.props.chooseHandler(this.props.data.id);
    }

    removeHandler(e: React.MouseEvent) {
        e.stopPropagation();
        this.props.removeHandler(this.props.data.id);
    }

    render() {
        const {data} = this.props;

        const classes = classNames({
            'post': true,
            'post_checked': this.props.isChecked,
        });

        return (
            <div className={classes} onClick={this.clickHandler}>
                <div className="post__inner">
                    <div className="post__title-box">
                        <h3 className="post__title title">{data.description}</h3>
                        <span className="post__comment-count">{this.props.comments_count}</span>
                    </div>
                    <button className="post__remove-btn btn-ordinary btn-ordinary_remove" onClick={this.removeHandler}>Delete</button>
                </div>
            </div>
        )
    }
}
