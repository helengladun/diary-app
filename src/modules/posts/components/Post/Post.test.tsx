import React from 'react';
import {shallow} from 'enzyme';
import {Post} from "./Post";
import {IPost} from "../../../shared/models/IPost";

describe('Post component', () => {
    const data: IPost = {
        id: 10,
        description: 'some post',
    };

    const removeHandler = jest.fn();
    const chooseHandler = jest.fn();
    const wrapper = shallow(<Post data={data} removeHandler={removeHandler} chooseHandler={chooseHandler} comments_count={10} isChecked={false}/>);

    test('should render Post component correctly', () => {
        expect(wrapper).toMatchSnapshot();
    });

    test('should fire removeHandler on click button', () => {
        wrapper.find('button').simulate('click', {
            stopPropagation: () => {}
        });

        expect(removeHandler).toHaveBeenLastCalledWith(data.id);
    });

    test('should fire chooseHandler on click node', () => {
        wrapper.find('div').at(0).simulate('click');

        expect(chooseHandler).toHaveBeenLastCalledWith(data.id);
    });

});