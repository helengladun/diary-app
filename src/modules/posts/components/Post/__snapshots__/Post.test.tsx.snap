// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Post component should render Post component correctly 1`] = `ShallowWrapper {}`;
